plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -861,7 +861,12 @@ else if (strings.size() != 2) {
             String schema = strings.get(0);
             String name = strings.get(1);
             SchemaTableName schemaTableName = new SchemaTableName(schema, name);
-            if (!isTableCurrent(session, getTableHandle(session, schemaTableName), entry.getValue())) {
+            IcebergTableHandle tableHandle = getTableHandle(session, schemaTableName);
+
+            if (tableHandle == null) {
```

Comment:
It's better to throw `MaterializedViewNotFoundException` (does not exists, see `io.trino.spi.connector.ViewNotFoundException` how to make one)